strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-nuget-layout-provider/src/main/java/org/carlspring/strongbox/repository/NugetRepositoryFeatures.java

```diff
@@ -115,4 +147,70 @@ protected Configuration getConfiguration()
         return configurationManager.getConfiguration();
     }
 
+    @Component
+    public class RepositorySearchEventListener implements CommonEventListener<RemoteRepositorySearchEvent>
+    {
+
+        @Override
+        public void handle(RemoteRepositorySearchEvent event)
+        {
+            RepositorySearchRequest repositorySearchRequest = event.getEventData();
+            Map<String, String> coordinates = repositorySearchRequest.getCoordinates();
+            String packageId = coordinates.get(NugetArtifactCoordinates.ID);
+            String version = coordinates.get(NugetArtifactCoordinates.VERSION);
+
+            Expression filter = repositorySearchRequest.isStrict() ? createPackageEq(packageId, null) : null;
+            filter = createVersionEq(version, filter);
+            String searchTerm = !repositorySearchRequest.isStrict() ? packageId : null;
+
+            try
+            {
+                downloadRemoteFeed(repositorySearchRequest.getStorageId(), repositorySearchRequest.getRepositoryId(),
+                                   filter,
+                                   searchTerm, null, repositorySearchRequest.getSkip(),
+                                   repositorySearchRequest.getLimit());
+            }
+            catch (RepositoryInitializationException | ArtifactTransportException e)
+            {
+                logger.error(String.format("Failed to fetch Nuget remote feed [%s]",
+                                           repositorySearchRequest.getCoordinates()),
+                             e);
+            }
+        }
+
+    }
+
+    public static Expression createVersionEq(String version,
+                                             Expression filter)
+    {
+        if (version == null || version.trim().isEmpty())
+        {
+            return filter;
+        }
+        VersionEq versionEq;
+        try
+        {
+            versionEq = new VersionEq(Version.parse(version));
+            filter = filter == null ? versionEq : new AndExpression(filter, versionEq);
+        }
+        catch (NugetFormatException e)
+        {
+            logger.error(String.format("Failed to parse Nuget version [%s]", version), e);
+        }
+
+        return filter;
+    }
+
+    public static Expression createPackageEq(String packageId,
+                                             Expression filter)
+    {
+        if (packageId == null || packageId.trim().isEmpty())
```

Comment:
`StringUtils.isBlank`

Reply:
Yeah, that sounds like a good idea.

Reply:
see above
but if you insist I will use it

Reply:
I'm not insisting, just a suggestion. I think `StringUtils.isBlank` is really self-explained.